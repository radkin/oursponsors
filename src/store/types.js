export const GET_SENATORS = 'GET_SENATORS';
export const SENATORS_ERROR = 'SENATORS_ERROR'
