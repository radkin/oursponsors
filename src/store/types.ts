// Senate
export const GET_SENATORS = 'GET_SENATORS';
export const SENATORS_ERROR = 'SENATORS_ERROR';
export const GET_MINI_SENATORS = 'GET_MINI_SENATORS';
export const MINI_SENATORS_ERROR = 'MINI_SENATORS_ERROR';
// Congress
export const GET_CONGRESS = 'GET_CONGRESS';
export const CONGRESS_ERROR = 'CONGRESS_ERROR';
export const GET_MINI_CONGRESS = 'GET_MINI_CONGRESS';
export const MINI_CONGRESS_ERROR = 'MINI_CONGRESS_ERROR';
// Preferences
export const GET_PREFERENCES = 'GET_PREFERENCES';
export const PREFERENCES_ERROR = 'PREFERENCES_ERROR';
export const UPDATE_PREFERENCES = 'UPDATE_PREFERENCES';
export const UPDATE_PREFERENCES_ERROR = 'UPDATE_PREFERENCES_ERROR';
// Details
export const GET_SECTORS = 'GET_SECTORS';
export const SECTORS_ERROR = 'SECTORS_ERROR';
export const GET_CONTRIBUTORS = 'GET_CONTRIBUTORS';
export const CONTRIBUTORS_ERROR = 'CONTRIBUTORS_ERROR';
// User
export const GET_USER = 'GET_USER';
export const USER_ERROR = 'USER_ERROR';
export const UPDATE_USER = 'UPDATE_USER';
export const SET_GOOGLE_UID = 'SET_GOOGLE_UID';
export const GET_SPONSORS = 'GET_SPONSORS';
export const SPONSORS_ERROR = 'SPONSORS_ERROR';
export const GET_SENATOR_DETAILS = 'GET_SENATOR_DETAILS';
export const SENATOR_DETAILS_ERROR = 'SENATOR_DETAILS_ERROR';
export const GET_CONGRESS_DETAILS = 'GET_CONGRESS_DETAILS';
export const CONGRESS_DETAILS_ERROR = 'CONGRESS_DETAILS_ERROR';
