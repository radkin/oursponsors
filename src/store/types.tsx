// Senate
export const GET_SENATORS = 'GET_SENATORS';
export const SENATORS_ERROR = 'SENATORS_ERROR';
export const GET_SENATORS_BY_STATE = 'GET_SENATORS_BY_STATE';
export const SENATORS_BY_STATE_ERROR = 'GET_SENATORS_BY_STATE_ERROR';

// Congress
export const GET_CONGRESS = 'GET_CONGRESS';
export const CONGRESS_ERROR = 'CONGRESS_ERROR';
export const GET_CONGRESS_BY_STATE = 'GET_CONGRESS_BY_STATE';
export const CONGRESS_BY_STATE_ERROR = 'CONGRESS_BY_STATE_ERROR';
// Preferences
export const GET_PREFERENCES = 'GET_PREFERENCES';
export const PREFERENCES_ERROR = 'PREFERENCES_ERROR';
export const UPDATE_PREFERENCES = 'UPDATE_PREFERENCES';
