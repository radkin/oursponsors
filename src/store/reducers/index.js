import {combineReducers} from 'redux';
import senatorReducer from './senatorReducer';
import congressReducer from './congressReducer';
import preferencesReducer from './preferencesReducer';
import updatePreferencesReducer from './updatePreferencesReducer';
import sectorReducer from './sectorReducer';
import contributorReducer from './contributorReducer';
import userReducer from './userReducer';
import firestoreReducer from './firestoreReducer';

export default combineReducers({
  senatorsList: senatorReducer,
  congressList: congressReducer,
  preferencesList: preferencesReducer,
  updatePreferences: updatePreferencesReducer,
  sectorsList: sectorReducer,
  contributorsList: contributorReducer,
  userObject: userReducer,
  googleUid: firestoreReducer,
});
