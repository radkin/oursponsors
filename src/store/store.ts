import { createStore, applyMiddleware, Action, AnyAction } from "redux";
import thunk, {ThunkAction, ThunkDispatch} from 'redux-thunk';
import {composeWithDevTools} from 'redux-devtools-extension';

import rootReducer from './reducers';
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";

const initalState = {};

const middleware = [thunk];

const store = createStore(
  rootReducer,
  initalState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

/* Types */
export type AppDispatch = typeof store.dispatch;
export type ReduxState = ReturnType<typeof rootReducer>;
export type TypedDispatch = ThunkDispatch<ReduxState, any, AnyAction>;
export type TypedThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  ReduxState,
  unknown,
  AnyAction
>;

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const useTypedDispatch = () => useDispatch<TypedDispatch>();
export const useTypedSelector: TypedUseSelectorHook<ReduxState> = useSelector;

export default store;
