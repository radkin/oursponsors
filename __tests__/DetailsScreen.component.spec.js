jest.useFakeTimers();

import renderer from 'react-test-renderer';
import RepDetails from '../src/screens/repDetails';
import {Provider} from 'react-redux';
import {Provider as PaperProvider} from 'react-native-paper';

import store from '../src/store/store';
describe('DetailsScreen', () => {
  it('renders correctly', () => {

    const route = {
      key: 'Details-jLO2Y2EFGMPQTG_IyPCzQ',
      name: 'Details',
      params: {
        value: {
          api_uri:
            'https://api.propublica.org/congress/v1/members/B000825.json',
          at_large: false,
          contact_form: null,
          crp_id: 'N00045974',
          cspan_id: null,
          date_of_birth: '1986-12-15',
          district: '3',
          dw_nominate: null,
          facebook_account: null,
          fec_candidate_id: 'H0CO03165',
          first_name: 'Lauren',
          gender: 'F',
          geoid: '0803',
          google_entity_id: '/g/11j2v06x6p',
          govtrack_id: 456805,
          icpsr_id: null,
          id: 399,
          image_url:
            'https://theunitedstates.io/images/congress/original/B000825.jpg',
          in_office: false,
          last_name: 'Boebert',
          last_updated: '2022-12-31T00:00:00',
          leadership_role: null,
          middle_name: null,
          missed_votes: 24,
          missed_votes_pct: 2.41,
          next_election: '2022',
          ocd_id: 'ocd-division/country:us/state:co/cd:3',
          office: '1609 Longworth House Office Building',
          party: 'R',
          phone: '202-225-4761',
          pro_publica_id: 'B000825',
          rss_url: null,
          seniority: 2,
          short_title: 'Rep.',
          state: 'CO',
          suffix: null,
          title: 'Representative',
          total_present: 1,
          total_votes: 997,
          twitter_account: 'RepBoebert',
          url: 'https://boebert.house.gov',
          votes_against_party_pct: 25.21,
          votes_with_party_pct: 74.69,
          votesmart_id: null,
          youtube_account: null,
        },
      },
      path: undefined,
    };

    const navigation = {navigate: jest.fn()};

    const props = {
      navigation: navigation,
      route: route,
    };

    // ToDo:     TypeError: Cannot read properties of undefined (reading 'params')
    // 45 |   const value = props.route.params?.value;
    // Pretty sure this is a problem with the test!

    const tree = renderer
      .create(
        <Provider store={store}>
          <PaperProvider>
            <RepDetails props={props} />
          </PaperProvider>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
