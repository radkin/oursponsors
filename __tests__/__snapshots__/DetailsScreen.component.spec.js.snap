// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      >
        Unable to display profile
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
